src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,627 @@
+                    bestIt = cluster;
+                }
+            }
+            else
+            {
+                /**
+                 * @Todo
+                 * What do we loose by converting from double to long ?
+                 */
+                double sendBitrateBps = cluster.meanSize * 8 * 1000
+                        / cluster.sendMeanMs;
+                double recvBitrateBps = cluster.meanSize * 8 * 1000
+                        / cluster.recvMeanMs;
+                logger.warn( "Probe failed, sent at " + sendBitrateBps
+                        + " bps, received at " + recvBitrateBps
+                        + " bps. Mean send delta: " + cluster.sendMeanMs
+                        + " ms, mean recv delta: " + cluster.recvMeanMs
+                        + " ms, num probes: " + cluster.count);
+                break;
+            }
+        }
+        return bestIt;
+    }
+
+    probeResult processClusters(long nowMs)
+    {
+        synchronized (critSect) {
+            List<Cluster> clusters = new ArrayList<Cluster>();
+            clusters = computeClusters(clusters);
+            if (clusters.isEmpty()) {
+                // If we reach the max number of probe packets and still
+                // have no clusters,
+                // we will remove the oldest one.
+                if (probes_.size() >= kMaxProbePackets)
+                    probes_.get(0);
+                return probeResult.kNoUpdate;
+            }
+
+            Cluster bestProbe = findBestProbe(clusters);
+            if (bestProbe != null) {
+                int probeBitrateBps =
+                        Math.min(bestProbe.getSendBitrateBps(), bestProbe
+                                .getRecvBitrateBps());
+                // Make sure that a probe sent on a lower bitrate
+                // than our estimate can't
+                // reduce the estimate.
+                if (isBitrateImproving(probeBitrateBps)) {
+                    logger.warn("Probe successful, sent at "
+                            + bestProbe.getSendBitrateBps() +
+                            " bps, received at "
+                            + bestProbe.getRecvBitrateBps()
+                            + " bps. Mean send delta: " + bestProbe.sendMeanMs
+                            + " ms, mean recv delta: " + bestProbe.recvMeanMs
+                            + " ms, num probes: " + bestProbe.count);
+                    remoteRate.setEstimate(probeBitrateBps, nowMs);
+                    return probeResult.kBitrateUpdated;
+                }
+            }
+            // Not probing and received non-probe packet,
+            // or finished with current set  of probes.
+            if (clusters.size() >= kExpectedNumberOfProbes)
+                probes_.clear();
+            return probeResult.kNoUpdate;
+        }
+    }
+
+    boolean isBitrateImproving(int newBitrateBps)
+    {
+        synchronized (critSect)
```

Comment:
Am using `synchronized` as the equivalent of `EXCLUSIVE_LOCKS_REQUIRED(&crit_)` in line 111 of [webrtc absSendTime.h](https://chromium.googlesource.com/external/webrtc/+/master/webrtc/modules/remote_bitrate_estimator/remote_bitrate_estimator_abs_send_time.cc) for  `isBitrateImproving`